### PR TITLE
NFS support for default storage

### PR DESCRIPTION
**Pull request by hyperbolic2346**
**What this PR does / why we need it**:
Adds NFS support for kubernetes-worker charm allowing default storage for NFS charms.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Added support for NFS relations on kubernetes-worker charm.
```

### CODE REVIEW

File: cluster/juju/layers/kubernetes-worker/layer.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ includes:
   - 'interface:kube-dns'
   - 'interface:kube-control'
   - 'interface:aws'
+  - 'interface:mount'
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-worker/layer.yaml]: Can you get interface-mount added to https://github.com/juju/layer-index? We shouldn't land this PR until it's in there.

Comment (hyperbolic2346) [reply cluster/juju/layers/kubernetes-worker/layer.yaml]: Done: https://github.com/juju/layer-index/pull/27

File: cluster/juju/layers/kubernetes-worker/templates/nfs-provisioner.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: nfs-client-provisioner
+    spec:
+      containers:
+        - name: nfs-client-provisioner
+          image: quay.io/external_storage/nfs-client-provisioner:latest
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-worker/templates/nfs-provisioner.yaml]: This is an amd64 image, meaning the NFS relation won't work on arm64, s390x, or others.

Let me think about this some more. I'll come back to this PR tomorrow.

Comment (hyperbolic2346) [reply cluster/juju/layers/kubernetes-worker/templates/nfs-provisioner.yaml]: Good point. I see an arm version, but nothing for s390x or others.. 

quay.io/external_storage/nfs-client-provisioner-arm:latest

**Comment by Cynerva** [comment]
/lgtm

Thanks @hyperbolic2346

File: cluster/juju/layers/kubernetes-worker/templates/nfs-provisioner.yaml (hunk 2 above)
Comment (Cynerva) [reply cluster/juju/layers/kubernetes-worker/templates/nfs-provisioner.yaml]: Discussed this offline, we settled on just shipping amd64 support for now.

**Pull request merged**